### ISSUE TITLE
Create Content Type for Index Page

### ISSUE THREAD

**Issue by kristianserrano** [emory-libraries/web-redesign#170]
- [x] Template
- [x] Data Definition
- [x] Metadata Set
- [x] Configuration Set
- [x] Base Asset
- [x] Asset Factory

**Comment by kristianserrano** [comment]
@laurenhamel this is ready for review on the dev side. It's pretty straightforward.

**Comment by laurenhamel** [comment]
@kristianserrano, this looks good. I went ahead and made one minor fix because the link description option wasn't showing for external links. Moving this into "To Accept."

**Comment by laurenhamel** [comment]
@nikdragovic, this is ready for your acceptance and should be available for content editors to start populating content into this page type.

**Comment by nikdragovic** [comment]
@laurenhamel Excellent, thank you so much. I'm going to try to start populating some pages today. Are you aware if there are currently pages on staging that are on this template? 

It looks like most of the pages marked Index in the site map are actually Landing in Cascade right now. If there are no Index pages, I'll just start from scratch with testing.

**Comment by laurenhamel** [comment]
I think @kristianserrano just created this page type last week, so there shouldn't be any content just yet that's using it. Feel free to start drafting some pages and/or migrating content from the Landing pages. Ideally, we could at least get a test page created today that I could use while building out the index template pattern.

**Comment by nikdragovic** [comment]
@laurenhamel Thanks for clarifying. I didn't read the issue carefully and thought that this was the template as opposed to just the content type in Cascade. I'll send acceptance ASAP.

**Comment by nikdragovic** [comment]
@laurenhamel Thanks for notifying me to take a look at this. I found an issue. I started a new Tools page here:

https://cascade.emory.edu/entity/open.act?id=f69b0cdc0ae720984942c44cb1397c60&type=page

There is an imposed limit of 8 links. We are going to need at least 13, based on the site map. 

Here is a Tools page Colin created using the Landing template:

https://cascade.emory.edu/entity/open.act?id=21ff431a0ae7209838e2f3a151d69043&type=page

https://staging.libraries.emory.edu/tools/index

It doesn't look too suitable for this content, but just wanted to let you know about the seeming duplicate.

The page I created has a mix of internal and external links; I varied in some places whether I was using a default description/label or override. I think there are some Tools pages created that need to be deleted in favor of an external link instead.

Let me know if this makes sense? I realize you'll need to talk to @kristianserrano

**Comment by laurenhamel** [comment]
@nikdragovic, I think that makes sense, and I agree that there shouldn't be an upper limit imposed on the number of pages that can be linked to. The lower limit, though, I think makes sense in terms of the design. I also took a second look at the data definition for this page, and I think we're also missing fields for images, which wouldn't necessarily be required but still an option.

**Comment by laurenhamel** [comment]
I'm moving this back to "In Progress" for now. Some additional discussion may be necessary to have with @kristianserrano, but I'm going to take this over and try to work on some of it since it's a blocker for the index template pattern.

**Comment by nikdragovic** [comment]
Thanks @laurenhamel

**Comment by laurenhamel** [comment]
@nikdragovic, I updated the data definition in Cascade to reflect the changes we discussed, and I went ahead and finished up the index template pattern based on these changes. Let me know if you run into any other issues. Otherwise, I think this should be good for acceptance.

**Comment by nikdragovic** [comment]
@laurenhamel Thanks, I'll take another run at it.

**Comment by nikdragovic** [comment]
@laurenhamel Do you know how I can tell the character limit for a custom
label or description before it gets truncated?

**Comment by laurenhamel** [comment]
@nikdragovic Are you talking about the custom link labels and descriptions? I don't think there are maximums set within the DD. As far as the tile pattern, though, there's only an imposed limit of 300 characters on the description but none on the label at present.

**Comment by nikdragovic** [comment]
@laurenhamel Thanks, that is helpful. It's not easy to tell on the Cascade
side, so I'll either have to inform folks about the limits or rely more on
a trial and error process to avoid truncation on pages like Tools.

**Comment by laurenhamel** [comment]
@nikdragovic, this is now ready for acceptance along with the Index page's template, so you should be able to run a full QA test on this DD.

**Comment by nikdragovic** [comment]
@laurenhamel @kristianserrano Here is the index page I was putting test
content into earlier this week: http://qa.libraries.emory.edu/tools1

Clearly, there are some place where I just wanted to see how extreme
numbers of characters were treated.

One question I have here is on the link, such as for discoverE here:
https://cascade.emory.edu/entity/open.act?id=f69b0cdc0ae720984942c44cb1397c60&type=page

There is an override in place to create a custom link description and
label. It appears that the description in Cascade goes where I'd expect the
link label to go. I'm not sure where the entered/intended link label is
going. This caused some stakeholder confusion on the landing page template
as well; Gretchen was confused about the /about/index page.

**Comment by laurenhamel** [comment]
@nikdragovic, this has been fixed for the index page. I went ahead and forced a reindex as well, so you should be seeing the correct results now in `qa`. As far as the landing page goes, I will have to check and see what's going on there since it's data definition is a bit different.

**Comment by laurenhamel** [comment]
@nikdragovic, I pinpointed the issue in the landing page, so I went ahead and pushed that fix out to `qa` as well. Part of the issue was that the conditional checks we were using for see if any link customizations should be applied were still being interpreted as truthy although we were expected it to be falsey (due to PHP type conversions when doing comparison). I'm looping @kristianserrano in because this will also impact other existing and pending templates, and we will need to update accordingly. I'm going to submit a new issue to capture this finding.
